corehq/apps/app_manager/suite_xml/features/case_tiles.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+from collections import namedtuple
+from eulxml.xmlmap.core import load_xmlobject_from_string
+from memoized import memoized
+from xml.sax.saxutils import escape
+
+from corehq.apps.app_manager import id_strings
+from corehq.apps.app_manager.exceptions import SuiteError
+from corehq.apps.app_manager.suite_xml.xml_models import Detail, XPathVariable
+
+
+CaseTileTemplateConfig = namedtuple('CaseTileTemplateConfig', ['slug', 'filename', 'fields'])
```

Comment:
I've come to prefer dataclasses for this kind of thing since they are about as simple to construct and provide better affordances in IDEs:

```python
DIRNAME = Path(__file__).parent / "case_tile_templates"

@dataclass
class CaseTileTemplateConfig:
    slug: str
    filename: str
    fields: List[str]

    @property
    def filepath(self):
        return DIRNAME / self.filename
```

You can also then create the objects directly from the loaded JSON data: `CaseTileTemplateConfig(**data)`

Reply:
that's much nicer, thank you - 61290bebaf1ab0314f7ce18170e981994b7cbe09